### DISCUSSION TITLE
Pager bug with data filter

### DISCUSSION THREAD

**Discussion by mpercec** [croz-ltd/tiller#177] [Q&A] [answered]
## Basic information

* Tiller version: 1.7.3
* Modules: -
* Link to the complete executable reproducer if available (e.g. GitHub Repo): -

## Additional information
- none

## Bug description
The pager below the table is disturbed, and starts showing negative page numbers: -7, -6, -5, etc.

### Steps play it
1. set the filter so that several pages of results are displayed
2. go to the last page
3. make additional filtering that will reduce the number of results
4. start filtering

### Expected behavior
The most logical thing would be to reset everything after filtering, i.e. to return to the first page.

### Actual behavior
It is described in the bug description.

**Comment by fbeceic** [comment, marked as answer]
This addition seems simple, but there are a lot of use cases to consider - filtering, sorting, deleting/adding rows, etc.

We'll see if we can elegantly cover all these use cases while maintaining functionality. 

In the meantime, if you're using the `useLocalPagination` hook and want the pagination to switch to the first page when filtering data resort to something like this:

```
useEffect(() => {
    paginationHook.onPageChange(0);
}, [dataToShow]) 
```
